src/main/java/core/basesyntax/Square.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax;
+
+public class Square extends Figure {
+    private double side;
+
+    public Square(Color color, double side) {
```

Comment:
Don't pass expressions in the constructor. Create separate variables and pass them on for better code readability. For example, if you're generating a random side length, do that in the FigureSupplier and pass the value here.